pandas/io/formats/info.py

```diff
@@ -41,22 +44,26 @@ def _put_str(s: Union[str, Dtype], space: int) -> str:
 
 def _get_ids_and_dtypes(data: FrameOrSeries) -> Tuple["Index", "Series"]:
     """
-    Get DataFrame's columns and dtypes.
+    Get DataFrame's columns (or Series' name) and dtypes.
 
     Parameters
     ----------
-    data : DataFrame
+    data : DataFrame or Series
         Object that `info` was called on.
 
     Returns
     -------
     ids : Index
-        DataFrame's columns.
+        DataFrame's columns or Series' name.
     dtypes : Series
-        Dtype of each of the DataFrame's columns.
+        Dtype of each of the DataFrame's columns or the Series' dtype.
     """
-    ids = data.columns
-    dtypes = data.dtypes
+    if isinstance(data, ABCDataFrame):
+        ids = data.columns
+        dtypes = data.dtypes
+    else:
+        ids = Index([data.name])
+        dtypes = Index([data.dtypes])
```

Comment:
Is there a particular reason for constructing an Index here? If not I think can use the machinery in pandas.core.common to convert to a list

Reply:
@WillAyd it's just to make typing easier. Would you prefer if it was a list then there was an assertion to help mypy at the lines
```python
    if isinstance(data, ABCDataFrame):
        counts = dtypes.value_counts().groupby(lambda x: x.name).sum()
```
?

Reply:
Actually, if I just use `data._constructor` then `dtypes` will be a Series in all cases, which means:

- no need to use `Union` in the return annotation
- no need to add extra assertions to help mypy

Reply:
Ah OK I see - I think this is OK then. Thanks for walking me through